scapy/arch/windows/__init__.py

```diff
@@ -67,16 +69,58 @@ def _encapsulate_admin(cmd):
     # To get admin access, we start a new powershell instance with admin rights, which will execute the command
     return "Start-Process PowerShell -windowstyle hidden -Wait -Verb RunAs -ArgumentList '-command &{%s}'" % cmd
 
+class _PowershellManager(Thread):
+    """Used only when scapy boots. Instance used to send multiple
+    commands on the same Powershell process. Skips the powershell
+    booting time to speed up scapy's boot"""
+    def __init__(self):
+        self.process = sp.Popen([conf.prog.powershell, "-NoLogo", "-NonInteractive", "-Command", "-"], # Start & redirect input
+                         stdout=sp.PIPE,
+                         stdin=sp.PIPE,
+                         stderr=sp.STDOUT,
+                         universal_newlines=True)
+        self.buffer = []
+        self.running = True
+        Thread.__init__(self)
+        self.start()
+
+    def run(self):
+        while self.running:
+            self.buffer.append(self.process.stdout.readline())
+
+    def query(self, cmd):
+        self.buffer = []
+        query = cmd + "; echo scapy_end\n"
+        self.process.stdin.write(query)
+        self.process.stdin.flush()
+        while self.running:
+            if self.buffer and len(self.buffer) >= 1:
+                if "scapy_end\n" in self.buffer:
+                    break
+        self.buffer.remove("scapy_end\n")
+        return self.buffer[1:]
+
+    def close(self):
+        self.running = False
+        try:
+            self.process.stdin.write("exit\n")
+            self.process.terminate()
+        except:
```

Comment:
Any way to use more specific except clause?

Reply:
There are many possible crashes, depending on if the process was already down, if only the stdin was closed...